index.js

```diff
@@ -71,16 +85,24 @@ function SonicBoom (fd, minLength, sync) {
   this.release = (err, n) => {
     if (err) {
       if (err.code === 'EAGAIN') {
-        // Let's give the destination some time to process the chunk.
-        // This error code should not happen in sync mode, because it is
-        // not using the underlining operating system asynchronous functions.
-        setTimeout(() => {
-          fs.write(this.fd, this._writingBuf, 'utf8', this.release)
-        }, 100)
-        return
+        if (this.sync) {
+          try {
+            sleep(100)
+            this.release(undefined, 0)
```

Comment:
`this.release(undefined, 0)` triggers the chunk that failed to be written to be written again. 

Calling `fs.writeSync(this.fd, this._buf)` is _not_ going to cause the buffer that errored to be written, but rather it will trigger the write of the rest of the chunks twice, because they will be written again at the next write.

Reply:
You're right, sorry, It was my mistake while reading and analysing release function.